icu4c/source/common/brkiter.cpp

```diff
@@ -204,28 +210,34 @@ BreakIterator::getAvailableLocales(int32_t& count)
 //
 //-------------------------------------------
 
-BreakIterator::BreakIterator()
+BreakIterator::BreakIterator() : actualLocale(new CharString()), validLocale(new CharString), requestLocale(new CharString())
 {
-    *validLocale = *actualLocale = *requestLocale = 0;
 }
 
-BreakIterator::BreakIterator(const BreakIterator &other) : UObject(other) {
-    uprv_strncpy(actualLocale, other.actualLocale, sizeof(actualLocale));
-    uprv_strncpy(validLocale, other.validLocale, sizeof(validLocale));
-    uprv_strncpy(requestLocale, other.requestLocale, sizeof(requestLocale));
+BreakIterator::BreakIterator(const BreakIterator &other) : UObject(other), actualLocale(new CharString()), validLocale(new CharString), requestLocale(new CharString()) {
+    UErrorCode status = U_ZERO_ERROR;
+    actualLocale->copyFrom(*(other.actualLocale), status);
```

Comment:
Before this line, please check if any of the three pointers is a nullptr. If so, then set the memory-allocation-error. Otherwise, copy.

Reply:
according https://en.cppreference.com/w/cpp/memory/new/operator_new if the new CharString() fail, it will not return nullptr but throw exception. So none of these three pointers could be a nullptr right?

Reply:
As far as I know, it is possible to build ICU4C in environments where operator `new` can return `nullptr` upon failure.